src/main/java/com/google/cloud/teleport/io/WindowedFilenamePolicy.java

```diff
@@ -38,60 +42,273 @@
  * writes.
  */
 @SuppressWarnings("serial")
-public class WindowedFilenamePolicy extends FilenamePolicy {
+@AutoValue
+public abstract class WindowedFilenamePolicy extends FilenamePolicy {
   /** The logger to output status messages to. */
   private static final Logger LOG = LoggerFactory.getLogger(WindowedFilenamePolicy.class);
-
-  private static final DateTimeFormatter YEAR = DateTimeFormat.forPattern("YYYY");
-  private static final DateTimeFormatter MONTH = DateTimeFormat.forPattern("MM");
-  private static final DateTimeFormatter DAY = DateTimeFormat.forPattern("dd");
-  private static final DateTimeFormatter HOUR = DateTimeFormat.forPattern("HH");
-  private static final DateTimeFormatter MINUTE = DateTimeFormat.forPattern("mm");
-  /** The filename baseFile. */
-  private final ValueProvider<String> outputDirectory;
-  /** The prefix of the file to output. */
-  private final ValueProvider<String> outputFilenamePrefix;
-  /** The filename suffix. */
-  private final ValueProvider<String> suffix;
-  /** The shard template used during file formatting. */
-  private final ValueProvider<String> shardTemplate;
-
+  
+  /** The default shard template used during file formatting. */
+  private static final String DEFAULT_SHARD_TEMPLATE = "W-P-SS-of-NN";
```

Comment:
Why do we need a default template given that we didnt have one earlier?

Reply:
Ah, I got these from the default values set in the Pubsub to Text template. I'll take them out.